app/controllers/devise_otp/devise/otp_tokens_controller.rb

```diff
@@ -23,7 +23,12 @@ def show
       # Updates the status of OTP authentication
       #
       def update
-        enabled = params[resource_name][:otp_enabled] == '1'
+        if params[resource_name].nil?
```

Comment:
Actually yes. I should've put a comment there but here's why. With Bootstrap JS, if a switch or fancy checkbox isn't checked, it doesn't get sent in the form. And if I remember correctly, because `:otp_enabled` doesn't get sent, `resource_name` also doesn't exist, causing an error to be through on line 29. Wrapping it in an `if` statement catches it, does that make sense?